backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/service/ItemStockService.java

```diff
@@ -0,0 +1,107 @@
+            return null;
+        }
+        return itemStockRepository.save(itemStock);
+    }
+
+    public ItemStock findById(ItemStock.Key key) {
+        return itemStockRepository.findById(key).orElse(null);
+    }
+
+    public List<ItemStock> findAll() {
+        return itemStockRepository.findAll();
+    }
+
+    private boolean validate(ItemStock itemStock) {
+        var key = itemStock.getKey();
+        try {
+            Objects.requireNonNull(key.getPartnerBpnl(), "Missing PartnerBpnl");
+            Objects.requireNonNull(key.getMaterialNumberCustomer(), "Missing materialNumberCustomer");
+            Objects.requireNonNull(key.getMaterialNumberSupplier(), "Missing materialNumberSupplier");
+            Objects.requireNonNull(key.getMaterialGlobalAssetId(), "Missing materialGlobalAssetId");
+            Objects.requireNonNull(key.getDirection(), "Missing direction");
+            Objects.requireNonNull(key.getSupplierOrderId(), "Missing supplierOrderId");
+            Objects.requireNonNull(key.getCustomerOrderId(), "Missing customerOrderId");
+            Objects.requireNonNull(key.getCustomerOrderPositionId(), "Missing customerOrderPositionId");
+            Objects.requireNonNull(key.getLocationBpna(), "Missing locationBpna");
+            Objects.requireNonNull(key.getLocationBpns(), "Missing locationBpns");
+            Objects.requireNonNull(itemStock.getMeasurementUnit(), "Missing measurementUnit");
+            Objects.requireNonNull(itemStock.getQuantityAmount(), "Missing quantityAmount");
+            Objects.requireNonNull(itemStock.getLastUpdatedOnDateTime(), "Missing lastUpdatedOnTime");
+            Partner partner =  partnerService.findByBpnl(key.getPartnerBpnl());
+            Objects.requireNonNull(partner, "Unknown partner: " + key.getPartnerBpnl());
+            Partner mySelf = partnerService.getOwnPartnerEntity();
+            Partner customer = key.getDirection() == DirectionCharacteristic.INBOUND ? mySelf : partner;
+            Partner supplier = customer == mySelf ? partner : mySelf;
+            var stockBpns = supplier.getSites().stream()
+                .filter(site -> site.getBpns().equals(key.getLocationBpns())).findFirst().orElse(null);
+            Objects.requireNonNull(stockBpns, "Unknown Bpns: " + key.getLocationBpns());
+            var stockBpna = supplier.getSites().stream().flatMap(site -> site.getAddresses().stream())
+                .filter(address -> address.getBpna().equals(key.getLocationBpna())).findFirst().orElse(null);
+            Objects.requireNonNull(stockBpna, "Unknown Bpna: " + key.getLocationBpna());
+            String ownMaterialNumber = mySelf == customer ? key.getMaterialNumberCustomer() : key.getMaterialNumberSupplier();
+            var materialPartnerRelation = mprService.find(ownMaterialNumber, partner.getUuid());
+            Objects.requireNonNull(materialPartnerRelation, "Missing MaterialPartnerRelation");
```

Comment:
please increase verbosity like "Missing MaterialPartnerRelationship between Partner A and Material B in the role of".

Also code is quite complex here -> can we bundle it a little more? Would a helper method be beneficial?

Reply:
IMHO thre readabilty it not so bad here, since every line contains a short text, which stated explicitly, what is being tested there. Splitting it up into various helper methods would do no good here either. As it currently is, you can see in one glance, what is being tested during the validation. That's far better than splitting this information into various bits and pieces across the Source code of the ItemStockService.